Formula/cling.rb

```diff
@@ -0,0 +1,38 @@
+class Cling < Formula
+  desc "The cling C++ interpreter"
+  homepage "https://root.cern.ch/cling"
+  url "http://root.cern.ch/git/cling.git",
+      :revision => "bfe05be443c3f16b2eff4094384f1bd22480f3bc"
+  version "0.4"
+
+  depends_on "cmake" => :build
+
+  resource "clang" do
+    url "http://root.cern.ch/git/clang.git", :branch => "cling-patches"
+  end
+
+  resource "llvm" do
+    url "http://root.cern.ch/git/llvm.git", :branch => "cling-patches"
+  end
+
+  def install
+    (buildpath/"src").install resource("llvm")
+    (buildpath/"src/tools/cling").install buildpath.children - [buildpath/"src"]
+    (buildpath/"src/tools/clang").install resource("clang")
+    cd buildpath/"src" do
```

Comment:
build dir can probably be at top level. no need to have it inside src

Reply:
What if `llvm` has some files which conflict with that of `cling` ?

Reply:
If you're in buildpath, and do `mkdir "build" do`, what would that conflict with?

Reply:
Sorry I meant in `buildpath`. 
If there is no `src` directory then `buildpath.install resource("llvm")` would extract files of llvm and source of `cling` would also be extracted in `buildpath`

Reply:
You can tell `cmake`  to look anywhere. The `build` directory doesn't need to be nested inside src.

Reply:
Ah nevermind. I misunderstood. You meant no need to have "build" inside "src"

Reply:
yup :)